feature-libs/cart/import-export/components/import-to-cart/import-to-cart.service.ts

```diff
@@ -162,19 +154,18 @@ export class ImportToCartService {
   /**
    * Emits `ProductImportInfo` on every added product success or failure
    */
-  protected getResults(cartId: string): Observable<ProductImportInfo> {
-    return this.placement$.pipe(
-      switchMap((placement) => {
-        switch (placement) {
-          case ImportCartRoutes.CART:
-          case ImportCartRoutes.SAVED_CARTS:
-          case ImportCartRoutes.SAVED_CART_DETAILS:
-          default: {
-            return this.getCartResults(cartId);
-          }
-        }
-      })
-    );
+  protected getResults(
+    cartType: CartTypes,
+    cartId: string
+  ): Observable<ProductImportInfo> {
+    switch (cartType) {
+      case CartTypes.ACTIVE_CART:
+      case CartTypes.NEW_SAVED_CART:
+      case CartTypes.EXISTING_SAVED_CART:
+      default: {
+        return this.getCartResults(cartId);
+      }
```

Comment:
I'm having hard time understanding why we need this switch-map.

Reply:
There will be a case for Quick order which need other logic, I will add todo here